internal/server/storage/drivers/driver_linstor_utils.go

```diff
@@ -0,0 +1,215 @@
+
+	if !util.PathExists(modulePath) {
+		return nil, fmt.Errorf("Could not determine DRBD module version: Module not loaded")
+	}
+
+	rawVersion, err := os.ReadFile(modulePath)
+	if err != nil {
+		return nil, fmt.Errorf("Could not determine DRBD module version: %w", err)
+	}
+
+	ver, err := version.Parse(strings.TrimSpace(string(rawVersion)))
+	if err != nil {
+		return nil, fmt.Errorf("Could not determine DRBD module version: %w", err)
+	}
+
+	return ver, nil
+}
+
+// resourceGroupExists returns whether the resource group associated with the current storage pool exists.
+func (d *linstor) resourceGroupExists() (bool, error) {
+	resourceGroup, err := d.getResourceGroup()
+	if err != nil {
+		return false, fmt.Errorf("Could not get resource group: %w", err)
+	}
+
+	if resourceGroup == nil {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// getResourceGroup fetches the resource group for the storage pool.
+func (d *linstor) getResourceGroup() (*linstorClient.ResourceGroup, error) {
+	d.logger.Debug("Fetching Linstor resource group")
+
+	// Retrieve the Linstor client
+	linstor, err := d.state.Linstor()
+	if err != nil {
+		return nil, fmt.Errorf("Could not load Linstor client: %w", err)
+	}
+
+	resourceGroupName := d.config[LinstorResourceGroupNameConfigKey]
+	resourceGroup, err := linstor.Client.ResourceGroups.Get(context.TODO(), resourceGroupName)
+	if errors.Is(err, linstorClient.NotFoundError) {
+		return nil, nil
+	} else if err != nil {
+		return nil, fmt.Errorf("Could not get Linstor resource group: %w", err)
+	}
+
+	return &resourceGroup, nil
+}
+
+// createResourceGroup creates a new resource group for the storage pool.
+func (d *linstor) createResourceGroup() error {
+	d.logger.Debug("Creating Linstor resource group")
+
+	// Retrieve the Linstor client
+	linstor, err := d.state.Linstor()
+	if err != nil {
+		return fmt.Errorf("Could not load Linstor client: %w", err)
+	}
+
+	placeCount, err := strconv.Atoi(d.config[LinstorResourceGroupPlaceCountConfigKey])
+	if err != nil {
+		return fmt.Errorf("Could not parse resource group place count property: %w", err)
+	}
+
+	resourceGroup := linstorClient.ResourceGroup{
+		Name:        d.config[LinstorResourceGroupNameConfigKey],
+		Description: "Resource group managed by Incus",
+		SelectFilter: linstorClient.AutoSelectFilter{
+			PlaceCount: int32(placeCount),
+		},
+	}
+
+	if d.config[LinstorResourceGroupStoragePoolConfigKey] != "" {
+		resourceGroup.SelectFilter.StoragePool = d.config[LinstorResourceGroupStoragePoolConfigKey]
+	}
+
+	err = linstor.Client.ResourceGroups.Create(context.TODO(), resourceGroup)
+	if err != nil {
+		return fmt.Errorf("Could not create Linstor resource group : %w", err)
+	}
+
+	return nil
+}
+
+// deleteResourceGroup deleter the resource group for the storage pool.
+func (d *linstor) deleteResourceGroup() error {
+	// Retrieve the Linstor client
+	linstor, err := d.state.Linstor()
+	if err != nil {
+		return fmt.Errorf("Could not load Linstor client: %w", err)
```

Comment:
This adds a prefix that does not really add any information to an already prefixed error